consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -1915,6 +1915,15 @@ impl<
                         Ok(verified) => {
                             if !verified {
                                 debug!(view = context.view, "proposal failed verification");
+                                 // Block the leader who proposed this invalid block
```

Comment:
Can you add a test for this? You should be able to modify the supervisor to coordinate failed verification for some leader and assert they end up blocked.